Rewrite Script Canvas section on Creating Custom Nodes (#1286)
o3de/o3de.org: content/docs/user-guide/scripting/script-canvas/creating-new-nodes.md

## Script events

You can create new sender and receiver event nodes in Script Canvas with the help of the **Asset Editor**. Configure a script event in Asset Editor by supplying the event name, parameters, and return value---similar to how you would define a function in any programming language, but no C++ code is required.

![Script event node example](/images/user-guide/scripting/script-canvas/script-event-node-example.png)

{{< note >}}
Script events are available to any O3DE scripting system, including Lua.
{{< /note >}}

Refer to the documentation on [Script events](/docs/user-guide/scripting/script-events/) for more information.

## Behavior context binding

To expose new C++ classes, methods, constants, data types, and events to Script Canvas, you can create script bindings to the behavior context. This is the most common method used to create new nodes for O3DE Gems and components.

![Example of node created from behavior context binding](/images/user-guide/scripting/script-canvas/behavior-context-node-example.png)

{{< note >}}
Script bindings are available to any O3DE scripting system that uses the behavior context, such as Lua.
{{< /note >}}

{{< note >}}

> No dependency on the Script Canvas Gem is required to create script bindings that reflect to the behavior context.

Author: Suggestion updated with your second option. That captures the essence of my intentions.

{{< /note >}}

Refer to the documentation on [Creating Script Canvas nodes from the behavior context](programmer-guide/behavior-context) in the Script Canvas Programmer Guide for more information.

## Custom nodes

> For the most control and flexibility when creating new nodes, including the ability to add new functionality, you can use the highly customizable node implementation mechanism called _nodeables_. Nodeables use XML configuration, Jinja templates, and an automatic C++ code generation tool to create custom Script Canvas nodes.

Reviewer: I would add., "This allows programmers to write classes in C++, using an SDK that will always be compatible with the Script Canvas Editor and runtime. The SDK will guarantee that the desired interface of the C++ class will be a useable node in the Editor, and an useable object during execution."

Author: Thanks! Let me know if my paraphrasing is accurate.

> For the most control and flexibility when creating new nodes, including the ability to add new functionality, you can use the highly customizable node implementation mechanism called _nodeables_. Nodeables use XML configuration, Jinja templates, and an automatic C++ code generation tool to create custom Script Canvas nodes. Using this mechanism, programmers can write classes in C++ using an SDK that is guaranteed to generate a class interface that provides a useable node in the Script Canvas Editor, and a useable object at runtime.

Reviewer 2: ```suggestion
For the most control and flexibility when creating new nodes, including the ability to add new functionality, you can use _nodeables_, a highly customizable node implementation mechanism. To create custom Script Canvas nodes, nodeables use XML configuration, Jinja templates, and an automatic C++ code generation tool. Programmers can use nodeables to write classes in C++ using an SDK that is guaranteed to generate a class interface that provides both a usable node in Script Canvas Editor and a usable object at runtime.
```
* Re-ordered some sentences for flow.
* Changed "useable" to the more common spelling "usable".

> ![Example of node created from nodeables](/images/user-guide/scripting/script-canvas/nodeable-node-example.png)

Reviewer 2: ```suggestion
![Example node created from nodeables](/images/user-guide/scripting/script-canvas/nodeable-node-example.png)
```
